From 53dd47fbf351bc75cafa036e0fbe73155cea4287 Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hauke.mehrtens@intel.com>
Date: Wed, 7 Nov 2018 10:11:09 +0100
Subject: [PATCH] DRVLIB_SW-938: Add dd_pool as parent of g_pon

The pon gate clock gets its clock from the dd_pool clock, the dd_pool clock is
activating the output 2 of the PLL3 which is then connected to the PON gate
clock. The PON gate clock provides the boot clock to the PON IP.
---
 drivers/clk/intel/clk-prx300.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/clk/intel/clk-prx300.c b/drivers/clk/intel/clk-prx300.c
index f903a03f0d4c..fd76afb93ee4 100644
--- a/drivers/clk/intel/clk-prx300.c
+++ b/drivers/clk/intel/clk-prx300.c
@@ -210,7 +210,7 @@ static const struct intel_clk_branch prx300_branch_clks[] __initconst = {
 		   G_DMA3_SHIFT, GATE_CLK_HW, 0),
 	INTEL_GATE(PRX300_GCLK_SWITCH, "g_switch", NULL, 0, CGU_GATE1,
 		   G_SWITCH_SHIFT, GATE_CLK_HW, 0),
-	INTEL_GATE(PRX300_GCLK_PON, "g_pon", NULL, 0, CGU_GATE1,
+	INTEL_GATE(PRX300_GCLK_PON, "g_pon", "dd_pool", 0, CGU_GATE1,
 		   G_PON_SHIFT, GATE_CLK_HW, 0),
 	INTEL_GATE(PRX300_GCLK_AON, "g_aon", NULL, 0, CGU_GATE1,
 		   G_AON_SHIFT, GATE_CLK_HW, 0),
